### ISSUE TITLE
Tunnel Config: Warp routing requires an update on every apply

### ISSUE THREAD

**Issue by joshblease** [cloudflare/terraform-provider-cloudflare#2577]
### Confirmation

- [ ] My issue isn't already found on the issue tracker. - https://github.com/cloudflare/terraform-provider-cloudflare/issues/2233
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.5.2
on darwin_arm64
+ provider registry.terraform.io/cloudflare/cloudflare v4.9.0
+ provider registry.terraform.io/hashicorp/random v3.5.1

### Affected resource(s)

cloudflare_tunnel_config

### Terraform configuration files

```hcl
resource "cloudflare_tunnel_config" "tunnel_config" {
  account_id = var.cloudflare_account_id
  tunnel_id  = var.cloudflare_tunnel_id
  config {
    warp_routing {
      enabled = true
    }
    ingress_rule {
      service = "http://10.10.0.10:8080"
      hostname = "foobar.com"
    }
    ingress_rule {
      service = "http_status:404"
    }
  }
}
```

### Link to debug output

https://gist.github.com/joshblease/73cd53fad6b725dc517877138aad48c7

### Expected output

I expect there to be no update due to the config not changing

### Actual output

Terraform always wants to set warp_routing enabled to true:
```
config {
  + warp_routing {
    + enabled = true
  }
  # (2 unchanged blocks hidden)
}

The tunnel, according to the cfd_tunnel endpoint, in the API has warp_routing set to false and is never updated.

### Steps to reproduce

Create a tunnel and config with warp_routing enabled.
Apply it to create the tunnel and then apply again to observe the enabling to be "applied" again but never is.

### Additional factoids

This doesn't appear to actually affect any functionality as the private network routing is working as expected. It's just frustrating to see a non-empty plan constantly.

### References

https://github.com/cloudflare/terraform-provider-cloudflare/issues/2233
https://github.com/pulumi/pulumi-cloudflare/issues/372

**Comment by jacobbednarz** [comment]
thanks for the issue and sorry to hear you're having issues. i've taken your reproduction but unable to reproduce the issue you're seeing.

```hcl
resource "cloudflare_tunnel" "example" {
  account_id = var.cloudflare_account_id
  name       = "my-example-tunnel"
  secret     = "AQIDBAUGBwgBAgMEBQYHCAECAwQFBgcIAQIDBAUGBwg="
}

resource "cloudflare_tunnel_config" "tunnel_config" {
  account_id = var.cloudflare_account_id
  tunnel_id  = cloudflare_tunnel.example.id
  config {
    warp_routing {
      enabled = true
    }
    ingress_rule {
      service  = "http://10.10.0.10:8080"
      hostname = "foobar.com"
    }
    ingress_rule {
      service = "http_status:404"
    }
  }
}
```

```
$ terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # cloudflare_tunnel.example will be created
  + resource "cloudflare_tunnel" "example" {
      + account_id   = "f037e56e89293a057740de681ac9abbe"
      + cname        = (known after apply)
      + id           = (known after apply)
      + name         = "my-example-tunnel"
      + secret       = (sensitive value)
      + tunnel_token = (sensitive value)
    }

  # cloudflare_tunnel_config.tunnel_config will be created
  + resource "cloudflare_tunnel_config" "tunnel_config" {
      + account_id = "f037e56e89293a057740de681ac9abbe"
      + id         = (known after apply)
      + tunnel_id  = (known after apply)

      + config {
          + ingress_rule {
              + hostname = "foobar.com"
              + service  = "http://10.10.0.10:8080"
            }
          + ingress_rule {
              + service = "http_status:404"
            }
          + warp_routing {
              + enabled = true
            }
        }
    }

Plan: 2 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cloudflare_tunnel.example: Creating...
cloudflare_tunnel.example: Creation complete after 2s [id=1060ca8c-3684-4fdf-bc57-800b4779901e]
cloudflare_tunnel_config.tunnel_config: Creating...
cloudflare_tunnel_config.tunnel_config: Creation complete after 1s [id=1060ca8c-3684-4fdf-bc57-800b4779901e]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
```

```
$ terraform apply
cloudflare_tunnel.example: Refreshing state... [id=1060ca8c-3684-4fdf-bc57-800b4779901e]
cloudflare_tunnel_config.tunnel_config: Refreshing state... [id=1060ca8c-3684-4fdf-bc57-800b4779901e]

No changes. Your infrastructure matches the configuration.

Terraform has compared your real infrastructure against your configuration and found no differences, so no changes are needed.

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.
```

are you sure you're able to replicate this on a new resource with the latest version?

**Comment by joshblease** [comment]
I can't reproduce this now using my own example @jacobbednarz, however I have recently migrated to Pulumi (which uses the TF provider) and I'm still facing the same issue for some tunnels but not all. It seems to only affect those that I have setup to route access applications, not sure if that's just a coincidence or not though.

I have even attempted to use the Cloudflare API to manually update the `warp-enabled` field and it doesn't modify the value either. I've reported this to Cloudflare so will get back to you once I have a response from them.
